### PR TITLE
Adding a timeout and retry to upload results step.

### PR DESCRIPTION
**Pull request by yusufm**
Upload metrics step has been failing sporadically (https://github.com/flutter/flutter/issues/82972) recently. Adding a timeout and retry to the call.

Timeout is set at 30secs. Successful completion of the step normally occurs within 17secs. And the security token is valid for 3minutes, which should be enough time for the retries to run as well.

### CODE REVIEW

File: dev/devicelab/lib/framework/cocoon.dart [hunk 1]
```diff
@@ -100,7 +100,11 @@ class Cocoon {
     resultsJson['TestFlaky'] = isTestFlaky ?? false;
     const List<String> supportedBranches = <String>['master'];
     if(supportedBranches.contains(resultsJson['CommitBranch'])) {
-      await _sendUpdateTaskRequest(resultsJson);
+      await retry(
+            () => _sendUpdateTaskRequest(resultsJson).timeout(const Duration(seconds: 30)),
```
Comment (godofredoc) [inline dev/devicelab/lib/framework/cocoon.dart:104]: Are 30 secs enough in slow network conditions?

Comment (CaseyHillers) [reply dev/devicelab/lib/framework/cocoon.dart:104]: Anecdotally, this operation should take <3 seconds (devicelab->cocoon->skia perf+datastore). If the network is slow on this HTTP request, there are likely larger issues with the recipe. Quickly retrying is preferred.

**Review by CaseyHillers — APPROVED** [review]

**Comment by CaseyHillers** [comment]
Please follow up with hixie regarding if this should have tests.

For infra repos, I'm okay skipping tests if we're adding a retry, but i'm not sure about the framework's policy.

**Comment by godofredoc** [comment]
> Please follow up with hixie regarding if this should have tests.
> 
> For infra repos, I'm okay skipping tests if we're adding a retry, but i'm not sure about the framework's policy.

That's a good point, let's add tests before landing this PR.

**Comment by yusufm** [comment]
Fixed sleep to be async, which correctly triggers the timeout. Added additional tests as well.

**Pull request merged**